prov/shm/src/smr_progress.c

```diff
@@ -587,10 +587,12 @@ static struct smr_pend_entry *smr_progress_ipc(struct smr_cmd *cmd,
 	if (cmd->msg.data.ipc_info.iface == FI_HMEM_ROCR) {
 		*total_len = 0;
 		ret = 0;
+		resp->status = SMR_STATUS_BUSY;
 		ipc_entry = smr_ipc_async_copy(ep,
 				(char*)ptr,
 				rx_entry, iov, iov_count, mr_entry, cmd, &ret);
-		resp->status = ret;
+		if (!ipc_entry)
+			resp->status = ret;
```

Comment:
The changes are pointing out that smr_ipc_async_copy() is odd, in that the return code is a parameter, rather than, well, a return code.  Can we change that function to return the return code and move ipc_entry as an output parameter?  That's more intuitive and makes the assignment of resp->status = ret more obvious.  Because on first reading, all that looked like it did was set resp->status = 0, since ret = 0 immediately above the call.